src/findcdn/cdnengine/detectcdn/cdn_check.py

```diff
@@ -0,0 +1,296 @@
+    # Check if we have initialized the ip list
+    if dom.ip_list is None:
+        dom.ip_list = list()
+    # Define our list of variant urls
+    dom_list: List[str] = [dom.url, "www." + dom.url]
+    # Lists to contain data from runs
+    return_codes = []
+    ip_list = []
+    # Iterate over domains for the ip addresses
+    for domain in dom_list:
+        try:
+            # Query the domain
+            response = query(domain)
+            # Assign any found IP addresses to the object
+            for ip_iter in response:
+                if (
+                    str(ip_iter.address) not in ip_list
+                    and str(ip_iter.address) not in dom.ip_list
+                ):
+                    ip_list.append(str(ip_iter.address))
+        except NoAnswer:
+            return_codes.append(1)
+        except NoNameservers:
+            return_codes.append(2)
+        except NXDOMAIN:
+            return_codes.append(3)
+        except Timeout:
+            return_codes.append(4)
+        except (socket.timeout, URLError, HTTPError):
+            pass
+
+    # Append all addresses into IP_list
+    for addr in ip_list:
+        dom.ip_list.append(addr)
+    # Return listing of error codes
+    return return_codes
+
+
+def cname(dom: Domain, timeout: int) -> List[int]:
+    """Collect CNAME records on domain."""
+    # Make sure we initialized the cname list
+    if dom.cnames is None:
+        dom.cnames = list()
+    # List of domains to check
+    dom_list = [dom.url, "www." + dom.url]
+    # Our codes to return
+    return_code = []
+    # Seutp resolver and timeouts
+    resolver = Resolver()
+    resolver.timeout = timeout
+    resolver.lifetime = LIFETIME
+    cname_query = resolver.query
+    # Iterate through all domains in list
+    for domain in dom_list:
+        try:
+            response = cname_query(domain, "cname")
+            dom.cnames = [record.to_text() for record in response]
+        except NoAnswer:
+            return_code.append(1)
+        except NoNameservers:
+            return_code.append(2)
+        except NXDOMAIN:
+            return_code.append(3)
+        except Timeout:
+            return_code.append(4)
+        except (socket.timeout, URLError, HTTPError):
+            pass
+    return return_code
+
+
+def https_lookup(dom: Domain, timeout: int, agent: str) -> int:
+    """Read 'server' header for CDN hints."""
+    # Check if we have initialized the headers list
+    if dom.headers is None:
+        dom.headers = list()
+    # List of domains with different protocols to check.
+    protocols = ["https://", "https://www."]
+    # Iterate through all protocols
+    for protocol in protocols:
+        try:
+            # Some domains only respond when we have a User-Agent defined.
+            req = request.Request(
+                protocol + dom.url, data=None, headers={"User-Agent": agent},
+            )
+            # Making the timeout 50 as to not hang thread.
```

Comment:
We probably don't need to say that the timeout is 50 because the timeout is passed in (also I don't think the default value is 50 either)

Reply:
Yeah it is not. I will go ahead and remove this.